hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmDBSnapshotInfoCodec.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.codec;
+
+import org.apache.hadoop.hdds.utils.db.Codec;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.SnapshotInfoEntry;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Codec to encode SnapshotInfo as byte array.
+ */
+public class OmDBSnapshotInfoCodec implements Codec<SnapshotInfo> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OmDBSnapshotInfoCodec.class);
+
+  @Override
+  public byte[] toPersistedFormat(SnapshotInfo object) throws IOException {
+    checkNotNull(object, "Null object can't be converted to byte array.");
+    return object.getProtobuf().toByteArray();
+  }
+
+  @Override
+  public SnapshotInfo fromPersistedFormat(byte[] rawData)
+      throws IOException {
+    checkNotNull(rawData, "Null byte array can't be converted to " +
+        "real object.");
+    return SnapshotInfo.getFromProtobuf(
+        SnapshotInfoEntry.parseFrom(rawData));
+  }
+
+  @Override
+  public SnapshotInfo copyObject(SnapshotInfo object) {
+    // Note: Not really a "copy". from OMDBSnapshotInfoCodec
+    return object;
```

Comment:
Seems odd that this is not a real copy.  OmBucketInfoCodec does it:

https://github.com/apache/ozone/blob/79a9d39da7f33e71bc00183e280105562354cca4/hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/codec/OmBucketInfoCodec.java#L54-L57

Reply:
As George pointed out, we can do a real copy here.

Reply:
Thought here was that we could do an implicit copy as there was no need for a 'deep' copy for the data structure as the `SnapshotInfo` consists of longs, ints and strings.  In the case of `OmKeyInfo` and `OmBucketInfo `objects, they include data structures within that warrant a deep copy, however this is not the case for `SnapshotInfo`, `OmDBTenantState`, `OmDBUserPrincipalInfo`.

I can create an explicit copy for the object.  Let me know if you think it is necessary.

Reply:
OmDBTenantState and OmDBUserPrincipalInfo both appear to be immutable, so it makes sense for those "copy's" to just return the original object.  SnapshotInfo isn't immutable, (even though its fields are.)

Reply:
Actually OmDBUserPrincipalInfo is not immutable.  So I would say that one is wrong too.